### PR TITLE
PWN-5520 - Setup and integrate AppsFlyer

### PR DESCRIPTION
**Pull request by davran1001**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5520

### CODE REVIEW

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/App.kt [hunk 1]
```diff
@@ -84,4 +87,12 @@ class App : Application() {
             setCustomKey("task_number", BuildConfig.TASK_NUMBER)
         }
     }
+
+    private fun setupAppsFlyer() {
+        val appsFlyer = AppsFlyerLib.getInstance().apply {
+            setDebugLog(BuildConfig.DEBUG)
+        }
+        appsFlyer.init(BuildConfig.appsFlyerKey, null, this@App)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/App.kt:95]: why not just `this`?

Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/App.kt:95]: please try to put appsflyer into a Service object like Intercom, we need abstraction here

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/App.kt:95]: also we can try and put the init block into a try catch - to remove the possibility of crashes if Appsflyer goes wrong

File: app/build.gradle [hunk 2]
```diff
@@ -309,6 +310,11 @@ dependencies {
     // Google Play Auth
     implementation 'com.google.android.gms:play-services-auth:20.3.0'
 
+    //AppsFlyer
+    implementation 'com.appsflyer:af-android-sdk:6.9.0'
+    implementation "com.android.installreferrer:installreferrer:2.2"
+
+
```
Comment (gslevinkov) [inline app/build.gradle:317]:
```suggestion
```

File: app/build.gradle [hunk 3]
```diff
@@ -31,6 +31,7 @@ def initApiKeys(defaultConfig) {
     defaultConfig.buildConfigField("String", "intercomApiKey", "${apiKeysProperties.getOrDefault('intercomApiKey', emptyString)}")
     defaultConfig.buildConfigField("String", "intercomAppId", "${apiKeysProperties.getOrDefault('intercomAppId', emptyString)}")
     defaultConfig.buildConfigField("String", "amplitudeKey", "${apiKeysProperties.getOrDefault('amplitudeKey', emptyString)}")
+    defaultConfig.buildConfigField("String","appsFlyerKey","${apiKeysProperties.getOrDefault('appsFlyerKey',emptyString)}")
```
Comment (gslevinkov) [inline app/build.gradle:34]: white spaces needed

**Review by gslevinkov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/App.kt (hunk 1 above)
Comment (davran1001) [reply app/src/main/java/org/p2p/wallet/App.kt:95]: Got you

File: app/src/main/java/org/p2p/wallet/App.kt [hunk 4]
```diff
@@ -44,6 +42,8 @@ class App : Application() {
         SolanjLogger.setLoggerImplementation(SolanajTimberLogger())
 
         appCreatedAction.invoke()
+
+        AppsFlyManager.install(this)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/App.kt:46]: AppsFlyManager  -> AppsflyerManager but it's better to stick with the current naming that we have - AppsflyerService

File: app/src/main/java/org/p2p/wallet/appsfly/AppsFlyManager.kt [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        val appsFlyer = AppsFlyerLib.getInstance().apply {
+            setDebugLog(BuildConfig.DEBUG)
+        }
+        appsFlyer.init(BuildConfig.appsFlyerKey, listener, application)
+        appsFlyer.start(application)
+    }
+
+    private object listener : AppsFlyerConversionListener {
+
+        override fun onConversionDataSuccess(p0: MutableMap<String, Any>?) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/appsfly/AppsFlyManager.kt:23]: you can rename p0 and other to other names without problem

File: app/src/main/java/org/p2p/wallet/appsfly/AppsFlyManager.kt [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+    fun install(application: Application) {
+        val appsFlyer = AppsFlyerLib.getInstance().apply {
+            setDebugLog(BuildConfig.DEBUG)
+        }
+        appsFlyer.init(BuildConfig.appsFlyerKey, listener, application)
+        appsFlyer.start(application)
+    }
+
+    private object listener : AppsFlyerConversionListener {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/appsfly/AppsFlyManager.kt:21]: fields should be places in the top of the class

File: app/src/main/java/org/p2p/wallet/appsfly/AppsFlyManager.kt [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+import android.app.Application
+import com.appsflyer.AppsFlyerConversionListener
+import com.appsflyer.AppsFlyerLib
+import org.p2p.wallet.BuildConfig
+import timber.log.Timber
+
+private const val TAG = "AppsFlyManager"
+
+object AppsFlyManager {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/appsfly/AppsFlyManager.kt:11]: object -> class

File: app/src/main/java/org/p2p/wallet/appsfly/AppsFlyManager.kt [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+    fun install(application: Application) {
+        val appsFlyer = AppsFlyerLib.getInstance().apply {
+            setDebugLog(BuildConfig.DEBUG)
+        }
+        appsFlyer.init(BuildConfig.appsFlyerKey, listener, application)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/appsfly/AppsFlyManager.kt:17]: better to pass it as a parameter

File: app/src/main/java/org/p2p/wallet/appsfly/AppsFlyManager.kt (hunk 6 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/appsfly/AppsFlyManager.kt:21]: and it's a inner object, not field. please change to a field please

**Review by gslevinkov — CHANGES_REQUESTED** [review]

**Review by 1sakov — APPROVED** [review]

File: app/build.gradle (hunk 3 above)
Comment (gslevinkov) [reply app/build.gradle:34]: no changes here, why?

**Review by gslevinkov — APPROVED** [review]

File: app/build.gradle (hunk 3 above)
Comment (davran1001) [reply app/build.gradle:34]: OoPs

Comment (davran1001) [reply app/build.gradle:34]: i will add it in next Pull request

**Pull request merged**